pkg/dependency/parser/python/uv/parse.go

```diff
@@ -0,0 +1,164 @@
+package uv
+
+import (
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	"github.com/samber/lo"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/dependency"
+	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	xio "github.com/aquasecurity/trivy/pkg/x/io"
+)
+
+type Lock struct {
+	Packages []Package `toml:"package"`
+}
+
+func (l Lock) packages() map[string]Package {
+	return lo.SliceToMap(l.Packages, func(pkg Package) (string, Package) {
+		return pkg.Name, pkg
+	})
+}
+
+func (l Lock) directDeps(root Package) map[string]struct{} {
+	deps := make(map[string]struct{})
+	for _, dep := range root.Dependencies {
+		deps[dep.Name] = struct{}{}
+	}
+	return deps
+}
+
+func prodDeps(root Package, packages map[string]Package) map[string]struct{} {
```

Comment:
I needed clarification on why this function could identify prod dependencies from the source code. Can you please add some comments?

Reply:
Added https://github.com/aquasecurity/trivy/pull/8080/commits/2a4674ab1270de6aeb4acae39f23a466ac6ae7da

Reply:
Are development dependencies listed in `[package.dev-dependencies]`? If so, we need to parse them for `--include-dev-deps`.

Reply:
We will do that in a separate PR.